tests/lib/utilities/DateParser.test.ts

```diff
@@ -1,144 +1,241 @@
+					const value = new DateParser(time, 'en-US');
+					validate(value);
+					expect(value.hour).toBe(8);
+					expect(value.minute).toBe(0);
+					expect(value.second).toBe(0);
+				});
+			});
+
+			describe('minutes', () => {
+				test.each(['20:05', '20:5', '20:05pm', '8:5pm'])('GIVEN %s THEN it resolves to 20:05', (time) => {
+					const value = new DateParser(time, 'en-US');
+					validate(value);
+					expect(value.hour).toBe(20);
+					expect(value.minute).toBe(5);
+					expect(value.second).toBe(0);
+				});
+
+				test.each(['08:05', '8:5', '20:05am', '8:5am'])('GIVEN %s THEN it resolves to 08:05', (time) => {
+					const value = new DateParser(time, 'en-US');
+					validate(value);
+					expect(value.hour).toBe(8);
+					expect(value.minute).toBe(5);
+					expect(value.second).toBe(0);
+				});
+			});
+
+			describe('seconds', () => {
+				test.each(['20:05:02', '20:5:2', '20:05:02pm', '8:5:2pm'])('GIVEN %s THEN it resolves to 20:05:02', (time) => {
+					const value = new DateParser(time, 'en-US');
+					validate(value);
+					expect(value.hour).toBe(20);
+					expect(value.minute).toBe(5);
+					expect(value.second).toBe(2);
+				});
+
+				test.each(['08:05:02', '8:5:2', '20:05:02am', '8:5:2am'])('GIVEN %s THEN it resolves to 08:05:02', (time) => {
+					const value = new DateParser(time, 'en-US');
+					validate(value);
+					expect(value.hour).toBe(8);
+					expect(value.minute).toBe(5);
+					expect(value.second).toBe(2);
+				});
+			});
 		});
 
-		test.each([
-			['DD/MM/YYYY', '10/03/2023'],
-			['DD-MM-YYYY', '10-03-2023'],
-			['DD.MM.YYYY', '10.03.2023'],
-			['DD/MM/YY', '10/03/23'],
-			['DD-MM-YY', '10-03-23'],
-			['DD.MM.YY', '10.03.23']
-		])('GIVEN %s with en-GB THEN it resolves to the correct date', (_format, date) => {
-			validate(new DateParser(date, 'en-GB'));
+		describe('datetime', () => {
+			test.each([
+				['YYYY/MM/DD and time', '2023/03/10 8pm', 'en-US'],
+				['MM/DD/YYYY and time with en-US locale', '03/10/2023 8pm', 'en-US'],
+				['DD/MM/YYYY and time with en-GB locale', '10/03/2023 8pm', 'en-GB']
+			] satisfies [description: string, input: string, locale: LocaleString][])(
+				'GIVEN %s THEN resolves to the correct datetime',
+				(_description, input, locale) => {
+					const value = new DateParser(input, locale);
+					expect(value.year).toBe(2023);
+					expect(value.month).toBe(3);
+					expect(value.day).toBe(10);
+					expect(value.hour).toBe(20);
+					expect(value.minute).toBe(0);
+					expect(value.second).toBe(0);
+					expect(value.valid).toBe(true);
+				}
+			);
 		});
 	});
 
-	describe('time', () => {
-		function validate(value: DateParser) {
-			expect(value.year).toBeNull();
-			expect(value.month).toBeNull();
-			expect(value.day).toBeNull();
-			expect(value.valid).toBe(true);
+	describe('normalize', () => {
+		beforeEach(() => {
+			vi.useFakeTimers({ now: new Date('2023-03-10T14:30:15.000Z') });
+		});
+
+		afterEach(() => {
+			vi.useRealTimers();
+		});
```

Comment:
```suggestion
		beforeAll(() => {
			vi.useFakeTimers({ now: new Date('2023-03-10T14:30:15.000Z') });
		});

		afterAll(() => {
			vi.useRealTimers();
		});
```

Surely this doesn't need an `each`